catalog/content/core/content-core-videothumbnailplugin/src/main/java/ddf/content/plugin/video/VideoThumbnailPlugin.java

```diff
@@ -0,0 +1,466 @@
+
+import ddf.catalog.data.Metacard;
+import ddf.content.data.ContentItem;
+import ddf.content.operation.CreateRequest;
+import ddf.content.operation.CreateResponse;
+import ddf.content.operation.UpdateRequest;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.operation.impl.CreateResponseImpl;
+import ddf.content.operation.impl.UpdateResponseImpl;
+import ddf.content.plugin.PluginExecutionException;
+import ddf.content.plugin.PostCreateStoragePlugin;
+import ddf.content.plugin.PostUpdateStoragePlugin;
+
+public class VideoThumbnailPlugin implements PostCreateStoragePlugin, PostUpdateStoragePlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VideoThumbnailPlugin.class);
+
+    private static final int FFMPEG_FILE_NUMBERING_START = 1;
+
+    private static final int THUMBNAIL_COUNT = 3;
+
+    private static final int ROUGH_MINIMUM_SECONDS_FOR_MULTIPLE_THUMBNAILS = 10;
+
+    private static final String SUPPRESS_PRINTING_BANNER_FLAG = "-hide_banner";
+
+    private static final String INPUT_FILE_FLAG = "-i";
+
+    private static final String OVERWRITE_EXISTING_FILE_FLAG = "-y";
+
+    private static final boolean DONT_HANDLE_QUOTING = false;
+
+    private static final int MAX_FFMPEG_PROCESSES = 4;
+
+    private final Semaphore limitFFmpegProcessesSemaphore;
+
+    private final String ffmpegPath;
+
+    public VideoThumbnailPlugin(final BundleContext bundleContext) throws IOException {
+        final String ffmpegFolderPath = FilenameUtils.concat(System.getProperty("ddf.home"),
+                "bin_third_party/ffmpeg");
+
+        if (SystemUtils.IS_OS_LINUX) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("linux/ffmpeg")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else if (SystemUtils.IS_OS_MAC) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("osx/ffmpeg")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg.exe");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("windows/ffmpeg.exe")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else if (SystemUtils.IS_OS_SOLARIS) {
+            ffmpegPath = FilenameUtils.concat(ffmpegFolderPath, "ffmpeg");
+            try (final InputStream inputStream = bundleContext.getBundle()
+                    .getEntry("solaris/ffmpeg")
+                    .openStream()) {
+                copyFFmpegBinary(inputStream);
+            }
+        } else {
```

Comment:
I see this as 2 cases.  1) Windows and 2) Everyone else.  In case #1, you'll store package the file in bin_third_party/ffmpeg/windows/ffmpeg.exe.  For case #2, store them under bin_third_party/ffmpeg/SystemUtilities.OS_ARCH/ffmeg.  That will eliminate most of this conditional and allow you to add binaries for other systems without changing code.